applications/nrf_desktop/src/modules/dfu_mcumgr.c

```diff
@@ -127,6 +134,7 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			cast_module_state_event(aeh);
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+#if defined CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
```

Comment:
L137-L168, replace with a single line:

```dfu_backend_init();```

And then define two variants of this function for each backend:

```
#if CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT
static void dfu_backend_init(void)
{
	if (!IS_ENABLE(CONFIG_DESKTOP_DFU_MCUMGR_MCUBOOT_DIRECT_XIP) {
		int err = boot_write_img_confirmed();

		if (err) {
			LOG_ERR("Cannot confirm a running image: %d", err);
		}
	}

	LOG_INF("MCUboot image version: %s", CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION);
}
#elif CONFIG_DESKTOP_DFU_BACKEND_SUIT
static void dfu_backend_init(void)
{
	/* Manifest class ID
	 * nRF Desktop uses default SUIT manifest file (default namespace and name).
	 */
	const suit_manifest_class_id_tmanifest_class_id = {{
		/* RFC4122 uuid5(uuid5(uuid.NAMESPACE_DNS, 'nordicsemi.com'),
		 *               'nRF54H20_sample_root')
		 */
		0x3f, 0x6a, 0x3a, 0x4d, 0xcd, 0xfa, 0x58, 0xc5,
		0xac, 0xce, 0xf9, 0xf5, 0x84, 0xc4, 0x11, 0x24
	}};
	unsigned int seq_num = 0;

	int err = suit_get_installed_manifest_info((
				suit_manifest_class_id_t *)&manifest_class_id,
				&seq_num, NULL, NULL, NULL, NULL);
	if (!err) {
		LOG_INF("SUIT image version: %d", seq_num);
	} else {
		LOG_ERR("suit_read_manifest_data failed (err: %d)", err);
	}
}
#else
#error "The DFU backend choice is not supported"
#endif /* CONFIG_DESKTOP_DFU_BACKEND_MCUBOOT */
```